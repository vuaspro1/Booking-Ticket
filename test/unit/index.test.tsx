/* import React from 'react'
import { render, fireEvent } from '../testUtils'
import Home from '../../pages/index'

describe('Home page', () => {
	it('matches snapshot', () => {
		const { asFragment } = render(<Home />, {})
		expect(asFragment()).toMatchSnapshot()
	})

	it('clicking button triggers alert', () => {
		const { getByText } = render(<Home />, {})
		window.alert = jest.fn()
		fireEvent.click(getByText('Test Button'))
		expect(window.alert).toHaveBeenCalledWith('With typescript and Jest')
	})
}) */

// import { toJS } from 'mobx'
import RootStore from '../../src/stores/RootStore'
import CartStore from '../../src/stores/cart.store'

const store = new RootStore()
const cartStore = store.cartStore

const newProduct = {
	id: 1,
	providerid: 1,
}

const newProduct1 = {
	id: 2,
	providerid: 1,
}

const newProduct2 = {
	id: 3,
	providerid: 2,
}

const newProduct3 = {
	id: 5,
	// providerid: 2,
}

describe('CartStore', () => {
	it('creates new CartStore', () => {
		expect(cartStore).toBeInstanceOf(CartStore)
	})

	it('CartStore addToCart', () => {
		cartStore.addToCart({ cartItem: newProduct, quantity: 100 })
		cartStore.addToCart({ cartItem: newProduct1, quantity: 1000 })
		cartStore.addToCart({ cartItem: newProduct2, quantity: 100 })
		cartStore.addToCart({ cartItem: newProduct3, quantity: 100 })
		expect(cartStore).toMatchSnapshot()
	})

	it('CartStore changeQuantityInCart', () => {
		cartStore.changeQuantityInCart({ id: 1, quantity: -10 })
		/* const quantityIds = {
			product: {
				'1': 90,
			},
			voucher: {},
		} */
		expect(cartStore.quantityIds.product[1]).toBe(90)
	})

	it('CartStore removeFromCart', () => {
		cartStore.removeFromCart({ id: 2 })
		expect(cartStore).toMatchSnapshot()
	})

	it('CartStore chooseToCheckoutCart', () => {
		cartStore.chooseToCheckoutCart({ providerId: 1, ids: [1] })

		expect(cartStore).toMatchSnapshot()
		/* const _checkoutCart = {
			'1': {
				product: [1],
				voucher: [],
			},
		}
		expect(cartStore.checkoutCart).toEqual(_checkoutCart) */
	})

	it('CartStore getCartItem', () => {
		const prod = cartStore.getCartItem({ id: 1 })
		expect(prod).toEqual(newProduct)
	})

	it('CartStore getItemQuantity', () => {
		const quantity = cartStore.getItemQuantity({ id: 1 })
		expect(quantity).toBe(90)
	})

	it('CartStore checkoutSuccess', () => {
		cartStore.checkoutSuccess({ providerId: 1, ids: [1] })
		expect(cartStore).toMatchSnapshot()
	})
})
