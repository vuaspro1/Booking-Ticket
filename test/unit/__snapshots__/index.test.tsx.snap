// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartStore CartStore addToCart 1`] = `
CartStore {
  "cartType": "product",
  "carts": Object {
    "product": Array [
      Object {
        "id": 1,
        "providerid": 1,
      },
      Object {
        "id": 2,
        "providerid": 1,
      },
      Object {
        "id": 3,
        "providerid": 2,
      },
      Object {
        "id": 5,
      },
    ],
    "voucher": Array [],
  },
  "checkoutCart": Object {
    "0": Object {
      "product": Array [],
      "voucher": Array [],
    },
  },
  "ids": Object {
    "product": Array [
      1,
      2,
      3,
      5,
    ],
    "voucher": Array [],
  },
  "providerIds": Object {
    "0": Object {
      "product": Array [
        5,
      ],
    },
    "1": Object {
      "product": Array [
        1,
      ],
    },
    "2": Object {
      "product": Array [
        3,
      ],
    },
  },
  "quantityCheck": Array [],
  "quantityIds": Object {
    "product": Object {
      "1": 100,
      "2": 1000,
      "3": 100,
      "5": 100,
    },
    "voucher": Object {},
  },
  "root": RootStore {
    "authStore": AuthStore {
      "auth": Object {},
      "chatId": null,
      "partnerInfo": Object {},
      "role": null,
      "root": [Circular],
      "state": "pending",
      "storeInfo": Object {},
      "token": null,
      "tokenChat": null,
    },
    "bankStore": BankStore {
      "bankCommons": Array [],
      "bankInfo": Object {},
      "bankPointReport": Array [],
      "bankSaleReport": Array [],
      "banks": Array [],
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "state": "pending",
    },
    "cardStore": CardStore {
      "cards": Array [],
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "state": "pending",
    },
    "cartStore": [Circular],
    "categoryStore": CategoryStore {
      "root": [Circular],
      "saleCategories": Array [],
      "saleCategoryInfo": Object {},
      "state": "pending",
    },
    "customerStore": CustomerStore {
      "customerInfo": Object {},
      "customerPointReport": Array [],
      "customerSaleReport": Array [],
      "customers": Array [],
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "state": "pending",
      "tokenUsedPoint": null,
    },
    "globalStore": GlobalStore {
      "root": [Circular],
      "state": "pending",
    },
    "loading": false,
    "orderStore": OrderStore {
      "hasEnd": false,
      "order": Object {},
      "orderInfo": Object {},
      "orderPointReport": Array [],
      "orderSaleReport": Array [],
      "orders": Array [],
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
        "status": "pending",
      },
      "root": [Circular],
      "state": "pending",
    },
    "productStore": ProductStore {
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "products": Array [],
      "root": [Circular],
      "state": "pending",
    },
    "providerStore": ProviderStore {
      "categories": Array [],
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "productInfo": Object {},
      "products": Array [],
      "providers": Array [],
      "root": [Circular],
      "state": "pending",
    },
    "reportStore": ReportStore {
      "hasEndReCharge": false,
      "hasEndRePoint": false,
      "hasEndReVoucher": false,
      "hasEndSaleSell": false,
      "hasEndWithDraw": false,
      "pagiReCharge": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "pagiRePoint": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "pagiReVoucher": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "pagiSaleSell": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "pagiWithDraw": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "paramReCharge": Object {
        "endTime": 1651424399999,
        "limit": 10,
        "skip": 0,
        "startTime": 1651338000000,
      },
      "paramRePoint": Object {
        "endTime": 1651424399999,
        "limit": 10,
        "skip": 0,
        "startTime": 1651338000000,
      },
      "paramReVoucher": Object {
        "endTime": 1651424399999,
        "limit": 10,
        "skip": 0,
        "startTime": 1651338000000,
      },
      "paramSaleSell": Object {
        "endTime": 1651424399999,
        "limit": 10,
        "skip": 0,
        "startTime": 1651338000000,
      },
      "paramWithDraw": Object {
        "endTime": 1651424399999,
        "limit": 10,
        "skip": 0,
        "startTime": 1651338000000,
      },
      "reChargePoints": Array [],
      "reportPoints": Object {},
      "reportSaleImport": Object {},
      "reportSaleSell": Object {},
      "reportSaleType": "sell",
      "reportVoucher": Object {},
      "revenues": Array [],
      "root": [Circular],
      "state": "pending",
      "topSales": Array [],
      "withDrawPoints": Array [],
    },
    "saleStore": SaleStore {
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "saleInfo": Object {},
      "saleInfos": Object {},
      "saleSummary": Object {},
      "sales": Array [],
      "state": "pending",
    },
    "shopStore": ShopStore {
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "shopInfo": Object {},
      "shops": Array [],
      "state": "pending",
    },
    "voucherStore": VoucherStore {
      "codeInfo": Object {},
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "state": "pending",
      "voucherInfo": Object {},
      "vouchers": Array [],
    },
    "workShift": WorkShiftStore {
      "root": [Circular],
      "state": "pending",
      "workshift": Object {},
    },
  },
  "state": "pending",
}
`;

exports[`CartStore CartStore checkoutSuccess 1`] = `
CartStore {
  "cartType": "product",
  "carts": Object {
    "product": Array [
      Object {
        "id": 1,
        "providerid": 1,
      },
      Object {
        "id": 3,
        "providerid": 2,
      },
      Object {
        "id": 5,
      },
    ],
    "voucher": Array [],
  },
  "checkoutCart": Object {
    "0": Object {
      "product": Array [],
      "voucher": Array [],
    },
  },
  "ids": Object {
    "product": Array [
      1,
      3,
      5,
    ],
    "voucher": Array [],
  },
  "providerIds": Object {
    "0": Object {
      "product": Array [],
    },
    "1": Object {
      "product": Array [
        1,
      ],
    },
    "2": Object {
      "product": Array [
        3,
      ],
    },
  },
  "quantityCheck": Array [],
  "quantityIds": Object {
    "product": Object {
      "1": 90,
      "3": 100,
      "5": 100,
    },
    "voucher": Object {},
  },
  "root": RootStore {
    "authStore": AuthStore {
      "auth": Object {},
      "chatId": null,
      "partnerInfo": Object {},
      "role": null,
      "root": [Circular],
      "state": "pending",
      "storeInfo": Object {},
      "token": null,
      "tokenChat": null,
    },
    "bankStore": BankStore {
      "bankCommons": Array [],
      "bankInfo": Object {},
      "bankPointReport": Array [],
      "bankSaleReport": Array [],
      "banks": Array [],
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "state": "pending",
    },
    "cardStore": CardStore {
      "cards": Array [],
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "state": "pending",
    },
    "cartStore": [Circular],
    "categoryStore": CategoryStore {
      "root": [Circular],
      "saleCategories": Array [],
      "saleCategoryInfo": Object {},
      "state": "pending",
    },
    "customerStore": CustomerStore {
      "customerInfo": Object {},
      "customerPointReport": Array [],
      "customerSaleReport": Array [],
      "customers": Array [],
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "state": "pending",
      "tokenUsedPoint": null,
    },
    "globalStore": GlobalStore {
      "root": [Circular],
      "state": "pending",
    },
    "loading": false,
    "orderStore": OrderStore {
      "hasEnd": false,
      "order": Object {},
      "orderInfo": Object {},
      "orderPointReport": Array [],
      "orderSaleReport": Array [],
      "orders": Array [],
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
        "status": "pending",
      },
      "root": [Circular],
      "state": "pending",
    },
    "productStore": ProductStore {
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "products": Array [],
      "root": [Circular],
      "state": "pending",
    },
    "providerStore": ProviderStore {
      "categories": Array [],
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "productInfo": Object {},
      "products": Array [],
      "providers": Array [],
      "root": [Circular],
      "state": "pending",
    },
    "reportStore": ReportStore {
      "hasEndReCharge": false,
      "hasEndRePoint": false,
      "hasEndReVoucher": false,
      "hasEndSaleSell": false,
      "hasEndWithDraw": false,
      "pagiReCharge": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "pagiRePoint": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "pagiReVoucher": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "pagiSaleSell": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "pagiWithDraw": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "paramReCharge": Object {
        "endTime": 1651424399999,
        "limit": 10,
        "skip": 0,
        "startTime": 1651338000000,
      },
      "paramRePoint": Object {
        "endTime": 1651424399999,
        "limit": 10,
        "skip": 0,
        "startTime": 1651338000000,
      },
      "paramReVoucher": Object {
        "endTime": 1651424399999,
        "limit": 10,
        "skip": 0,
        "startTime": 1651338000000,
      },
      "paramSaleSell": Object {
        "endTime": 1651424399999,
        "limit": 10,
        "skip": 0,
        "startTime": 1651338000000,
      },
      "paramWithDraw": Object {
        "endTime": 1651424399999,
        "limit": 10,
        "skip": 0,
        "startTime": 1651338000000,
      },
      "reChargePoints": Array [],
      "reportPoints": Object {},
      "reportSaleImport": Object {},
      "reportSaleSell": Object {},
      "reportSaleType": "sell",
      "reportVoucher": Object {},
      "revenues": Array [],
      "root": [Circular],
      "state": "pending",
      "topSales": Array [],
      "withDrawPoints": Array [],
    },
    "saleStore": SaleStore {
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "saleInfo": Object {},
      "saleInfos": Object {},
      "saleSummary": Object {},
      "sales": Array [],
      "state": "pending",
    },
    "shopStore": ShopStore {
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "shopInfo": Object {},
      "shops": Array [],
      "state": "pending",
    },
    "voucherStore": VoucherStore {
      "codeInfo": Object {},
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "state": "pending",
      "voucherInfo": Object {},
      "vouchers": Array [],
    },
    "workShift": WorkShiftStore {
      "root": [Circular],
      "state": "pending",
      "workshift": Object {},
    },
  },
  "state": "pending",
}
`;

exports[`CartStore CartStore chooseToCheckoutCart 1`] = `
CartStore {
  "cartType": "product",
  "carts": Object {
    "product": Array [
      Object {
        "id": 1,
        "providerid": 1,
      },
      Object {
        "id": 3,
        "providerid": 2,
      },
      Object {
        "id": 5,
      },
    ],
    "voucher": Array [],
  },
  "checkoutCart": Object {
    "0": Object {
      "product": Array [],
      "voucher": Array [],
    },
    "1": Object {
      "product": Array [
        1,
      ],
    },
  },
  "ids": Object {
    "product": Array [
      1,
      3,
      5,
    ],
    "voucher": Array [],
  },
  "providerIds": Object {
    "0": Object {
      "product": Array [],
    },
    "1": Object {
      "product": Array [
        1,
      ],
    },
    "2": Object {
      "product": Array [
        3,
      ],
    },
  },
  "quantityCheck": Array [],
  "quantityIds": Object {
    "product": Object {
      "1": 90,
      "3": 100,
      "5": 100,
    },
    "voucher": Object {},
  },
  "root": RootStore {
    "authStore": AuthStore {
      "auth": Object {},
      "chatId": null,
      "partnerInfo": Object {},
      "role": null,
      "root": [Circular],
      "state": "pending",
      "storeInfo": Object {},
      "token": null,
      "tokenChat": null,
    },
    "bankStore": BankStore {
      "bankCommons": Array [],
      "bankInfo": Object {},
      "bankPointReport": Array [],
      "bankSaleReport": Array [],
      "banks": Array [],
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "state": "pending",
    },
    "cardStore": CardStore {
      "cards": Array [],
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "state": "pending",
    },
    "cartStore": [Circular],
    "categoryStore": CategoryStore {
      "root": [Circular],
      "saleCategories": Array [],
      "saleCategoryInfo": Object {},
      "state": "pending",
    },
    "customerStore": CustomerStore {
      "customerInfo": Object {},
      "customerPointReport": Array [],
      "customerSaleReport": Array [],
      "customers": Array [],
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "state": "pending",
      "tokenUsedPoint": null,
    },
    "globalStore": GlobalStore {
      "root": [Circular],
      "state": "pending",
    },
    "loading": false,
    "orderStore": OrderStore {
      "hasEnd": false,
      "order": Object {},
      "orderInfo": Object {},
      "orderPointReport": Array [],
      "orderSaleReport": Array [],
      "orders": Array [],
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
        "status": "pending",
      },
      "root": [Circular],
      "state": "pending",
    },
    "productStore": ProductStore {
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "products": Array [],
      "root": [Circular],
      "state": "pending",
    },
    "providerStore": ProviderStore {
      "categories": Array [],
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "productInfo": Object {},
      "products": Array [],
      "providers": Array [],
      "root": [Circular],
      "state": "pending",
    },
    "reportStore": ReportStore {
      "hasEndReCharge": false,
      "hasEndRePoint": false,
      "hasEndReVoucher": false,
      "hasEndSaleSell": false,
      "hasEndWithDraw": false,
      "pagiReCharge": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "pagiRePoint": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "pagiReVoucher": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "pagiSaleSell": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "pagiWithDraw": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "paramReCharge": Object {
        "endTime": 1651424399999,
        "limit": 10,
        "skip": 0,
        "startTime": 1651338000000,
      },
      "paramRePoint": Object {
        "endTime": 1651424399999,
        "limit": 10,
        "skip": 0,
        "startTime": 1651338000000,
      },
      "paramReVoucher": Object {
        "endTime": 1651424399999,
        "limit": 10,
        "skip": 0,
        "startTime": 1651338000000,
      },
      "paramSaleSell": Object {
        "endTime": 1651424399999,
        "limit": 10,
        "skip": 0,
        "startTime": 1651338000000,
      },
      "paramWithDraw": Object {
        "endTime": 1651424399999,
        "limit": 10,
        "skip": 0,
        "startTime": 1651338000000,
      },
      "reChargePoints": Array [],
      "reportPoints": Object {},
      "reportSaleImport": Object {},
      "reportSaleSell": Object {},
      "reportSaleType": "sell",
      "reportVoucher": Object {},
      "revenues": Array [],
      "root": [Circular],
      "state": "pending",
      "topSales": Array [],
      "withDrawPoints": Array [],
    },
    "saleStore": SaleStore {
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "saleInfo": Object {},
      "saleInfos": Object {},
      "saleSummary": Object {},
      "sales": Array [],
      "state": "pending",
    },
    "shopStore": ShopStore {
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "shopInfo": Object {},
      "shops": Array [],
      "state": "pending",
    },
    "voucherStore": VoucherStore {
      "codeInfo": Object {},
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "state": "pending",
      "voucherInfo": Object {},
      "vouchers": Array [],
    },
    "workShift": WorkShiftStore {
      "root": [Circular],
      "state": "pending",
      "workshift": Object {},
    },
  },
  "state": "pending",
}
`;

exports[`CartStore CartStore removeFromCart 1`] = `
CartStore {
  "cartType": "product",
  "carts": Object {
    "product": Array [
      Object {
        "id": 1,
        "providerid": 1,
      },
      Object {
        "id": 3,
        "providerid": 2,
      },
      Object {
        "id": 5,
      },
    ],
    "voucher": Array [],
  },
  "checkoutCart": Object {
    "0": Object {
      "product": Array [],
      "voucher": Array [],
    },
  },
  "ids": Object {
    "product": Array [
      1,
      3,
      5,
    ],
    "voucher": Array [],
  },
  "providerIds": Object {
    "0": Object {
      "product": Array [],
    },
    "1": Object {
      "product": Array [
        1,
      ],
    },
    "2": Object {
      "product": Array [
        3,
      ],
    },
  },
  "quantityCheck": Array [],
  "quantityIds": Object {
    "product": Object {
      "1": 90,
      "3": 100,
      "5": 100,
    },
    "voucher": Object {},
  },
  "root": RootStore {
    "authStore": AuthStore {
      "auth": Object {},
      "chatId": null,
      "partnerInfo": Object {},
      "role": null,
      "root": [Circular],
      "state": "pending",
      "storeInfo": Object {},
      "token": null,
      "tokenChat": null,
    },
    "bankStore": BankStore {
      "bankCommons": Array [],
      "bankInfo": Object {},
      "bankPointReport": Array [],
      "bankSaleReport": Array [],
      "banks": Array [],
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "state": "pending",
    },
    "cardStore": CardStore {
      "cards": Array [],
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "state": "pending",
    },
    "cartStore": [Circular],
    "categoryStore": CategoryStore {
      "root": [Circular],
      "saleCategories": Array [],
      "saleCategoryInfo": Object {},
      "state": "pending",
    },
    "customerStore": CustomerStore {
      "customerInfo": Object {},
      "customerPointReport": Array [],
      "customerSaleReport": Array [],
      "customers": Array [],
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "state": "pending",
      "tokenUsedPoint": null,
    },
    "globalStore": GlobalStore {
      "root": [Circular],
      "state": "pending",
    },
    "loading": false,
    "orderStore": OrderStore {
      "hasEnd": false,
      "order": Object {},
      "orderInfo": Object {},
      "orderPointReport": Array [],
      "orderSaleReport": Array [],
      "orders": Array [],
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
        "status": "pending",
      },
      "root": [Circular],
      "state": "pending",
    },
    "productStore": ProductStore {
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "products": Array [],
      "root": [Circular],
      "state": "pending",
    },
    "providerStore": ProviderStore {
      "categories": Array [],
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "productInfo": Object {},
      "products": Array [],
      "providers": Array [],
      "root": [Circular],
      "state": "pending",
    },
    "reportStore": ReportStore {
      "hasEndReCharge": false,
      "hasEndRePoint": false,
      "hasEndReVoucher": false,
      "hasEndSaleSell": false,
      "hasEndWithDraw": false,
      "pagiReCharge": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "pagiRePoint": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "pagiReVoucher": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "pagiSaleSell": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "pagiWithDraw": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "paramReCharge": Object {
        "endTime": 1651424399999,
        "limit": 10,
        "skip": 0,
        "startTime": 1651338000000,
      },
      "paramRePoint": Object {
        "endTime": 1651424399999,
        "limit": 10,
        "skip": 0,
        "startTime": 1651338000000,
      },
      "paramReVoucher": Object {
        "endTime": 1651424399999,
        "limit": 10,
        "skip": 0,
        "startTime": 1651338000000,
      },
      "paramSaleSell": Object {
        "endTime": 1651424399999,
        "limit": 10,
        "skip": 0,
        "startTime": 1651338000000,
      },
      "paramWithDraw": Object {
        "endTime": 1651424399999,
        "limit": 10,
        "skip": 0,
        "startTime": 1651338000000,
      },
      "reChargePoints": Array [],
      "reportPoints": Object {},
      "reportSaleImport": Object {},
      "reportSaleSell": Object {},
      "reportSaleType": "sell",
      "reportVoucher": Object {},
      "revenues": Array [],
      "root": [Circular],
      "state": "pending",
      "topSales": Array [],
      "withDrawPoints": Array [],
    },
    "saleStore": SaleStore {
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "saleInfo": Object {},
      "saleInfos": Object {},
      "saleSummary": Object {},
      "sales": Array [],
      "state": "pending",
    },
    "shopStore": ShopStore {
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "shopInfo": Object {},
      "shops": Array [],
      "state": "pending",
    },
    "voucherStore": VoucherStore {
      "codeInfo": Object {},
      "hasEnd": false,
      "pagination": Object {
        "limit": 10,
        "loadMore": false,
        "page": 1,
        "skip": 0,
        "total": 0,
        "totalPage": 0,
      },
      "params": Object {
        "limit": 10,
        "skip": 0,
      },
      "root": [Circular],
      "state": "pending",
      "voucherInfo": Object {},
      "vouchers": Array [],
    },
    "workShift": WorkShiftStore {
      "root": [Circular],
      "state": "pending",
      "workshift": Object {},
    },
  },
  "state": "pending",
}
`;
